source/common/router/router.cc

```diff
@@ -519,17 +520,23 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
   // Fetch a connection pool for the upstream cluster.
   const auto& upstream_http_protocol_options = cluster_->upstreamHttpProtocolOptions();
 
-  if (upstream_http_protocol_options.has_value() &&
-      upstream_http_protocol_options.value().auto_sni()) {
+  if (upstream_http_protocol_options.has_value()) {
     const auto host_str = headers.Host()->value().getStringView();
     const auto parsed_authority = Http::Utility::parseAuthority(host_str);
     if (!parsed_authority.is_ip_address_) {
-      // TODO: Add SAN verification here and use it from dynamic_forward_proxy
-      // Update filter state with the host/authority to use for setting SNI in the transport
-      // socket options. This is referenced during the getConnPool() call below.
```

Comment:
Is this TODO still relevant? Is there cleanup we can do in the forward proxy code now with this code available?

Reply:
Ugh I forgot to add that part yet, but it will be in this PR. will update soon.